spinalcordtoolbox/registration/core.py

```diff
@@ -392,6 +392,19 @@ def register(src, dest, step, param):
             verbose=param.verbose,
         )
 
+    # deep learning registration using cascaded models
+    elif step.algo == 'dl':
+        # check if user provided a mask-- if so, inform it will be ignored
+        if fname_mask:
+            printv('\nWARNING: algo ' + step.algo + ' will ignore the provided mask.\n', 1, 'warning')
```

Comment:
Actually, on second thought, this is subtly different behavior vs. the stated purpose of `-m`:

```
  -m <file>             Mask that can be created with sct_create_mask to improve accuracy over region 
                        of interest. This mask will be used on the destination image. Example: mask.nii.gz
```

 `-m` isn't used by the deepreg method in this way, so it might be best to keep things simple...

Reply:
Yes it was a good idea but you are right `-m` is in fact used to focus the registration on a specific area, while our recommendations are mostly to reduce the images size and are not used to lead to more accurate registration results. So I agree with you, I think that we can't really use `-m` here.